### ISSUE TITLE
peerDependencies are missing in the next releases

### ISSUE THREAD

**Issue by briandipalma** [LegendApp/legend-state#237]
I'd like to test the next releases but the peerDependencies defined in master are missing.

**Comment by jmeistrich** [comment]
I think this should be fixed in next.41. I wanted to make the imports optional instead of being peerDependencies because they're not required for usage in all platforms. I think it's correct now. Please let me know if it's still not right :)

**Comment by briandipalma** [comment]
Nope, I get the same errors as before, react isn't installed. Optional provides no guarantee from what I can tell, so I guess pnpm doesn't install/create the symlinks in a pnpm workspace setup?

![image](https://github.com/LegendApp/legend-state/assets/1597820/f5672046-bdcc-4f40-9011-52572cc90255)

**Comment by briandipalma** [comment]
I'm also seeing this weird error now:

![image](https://github.com/LegendApp/legend-state/assets/1597820/5b6d8f95-5688-4a58-9f16-2d3dad528a4a)

**Comment by jmeistrich** [comment]
Oh it's react that's the issue? Do you have react installed and it's not resolving it? Or are you using it without react?

I haven't used pnpm so I'm not sure how it works. In npm/yarn/bun I believe it resolves dependencies correctly if they're installed even if not specified in the package. Does pnpm require the package to list its dependences? 

I don't want to make react a required dependency because the core should be usable without react, and I'd like to expand to other frameworks.

Do you (or anyone) know how to fix this correctly? One solution could be to make the react features a completely separate package that has react as a dependency, but I'm hoping there's an easier way?

**Comment by briandipalma** [comment]
No need to make React a required dependency. It should be just like it was before a peer dependency.

From https://classic.yarnpkg.com/lang/en/docs/dependency-types/#toc-peerdependencies

Peer dependencies are a special type of dependency that would only ever come up if you were publishing your own package.

Having a peer dependency means that your package needs a dependency that is the same exact dependency as the person installing your package. This is useful for packages like react that need to have a single copy of react-dom that is also used by the person installing it.

**Comment by IgorDePaula** [comment]
![Image](https://github.com/user-attachments/assets/bdcb9bce-ee5c-4cd6-8418-2bcb797d563a)
